### ISSUE TITLE
UglifyJsPlugin seems to remove a text type attribute of the input tag from html string that loaded by html-loader

### ISSUE THREAD

**Issue by drillbits** [webpack/webpack#752]
I built and uglified a source code that requires some html files using html-loder. webpack.optimize.UglifyJsPlugin seems to remove a text type attribute of the input tag from html string.

The default value of the type attribute is text. But the CSS selector such as `input[type="text"]` does not applied if the type attribute is removed.

My test repository is [here](https://github.com/drillbits/webpack-uglify-removes-input-type).

Now, I'm uglifying on another task using grunt-contrib-uglify for work arround.

**Comment by jhnns** [comment]
That's not actual an issue imho, but you can argue about the aggressiveness of the html minifier ^^. Removing the attribute to save space is correct, because the default is `text`... maybe you should adjust your css to match input fields without a type attribute.

**Comment by drillbits** [comment]
Unfortunately, I use third-party's css and it defines `input[type="text"]`...

I read `lib/optimize/UglifyJsPlugin.js` and noticed minimize is always `true`. So I create Pull request #795.

**Comment by bestguy** [comment]
Yeah this is really thrashing my HTML, at least with an Ractive project - it's skipping closing tags and breaking the output.

**Comment by sokra** [comment]
https://github.com/webpack/webpack/issues/283

This will change with webpack 2.0

**Comment by bestguy** [comment]
Awesome, thanks.  I was totally confused why this was happening in production!

**Comment by nicco** [comment]
looking forward to get this fixed....

**Comment by drillbits** [comment]
:+1:

**Comment by loddit** [comment]
:+1:

**Comment by elwynelwyn** [comment]
:+1:

**Comment by ufocoder** [comment]
To solve this problem set `minimize` = false for `loader` html. 
I spend half of my workday to find out it

**Comment by FTAndy** [comment]
@ufocoder Thanks. Save my time.

**Comment by pkantsedalov** [comment]
Hi guys.
Neither "minimize=false" nor "removeRedundantAttributes=false" works for me: **type="text"** disappears in each production build, where I use UglifyJsPlugin.

Here is my code:
``` javascript 
 html:   {
        test:    /\.html$/,
        exclude: /node_modules/,
        loader:  'html?minimize=false&removeRedundantAttributes=false'
    },
...
uglifyJs: () => {
        return new webpack.optimize.UglifyJsPlugin({
            mangle: false,
            compress: {
                warnings: false
            }
        });
    },
```

Versions:
**css-loader**: 0.23.0
**html-loader**: 0.4.0
**webpack**: 1.12.1

**Comment by pkantsedalov** [comment]
Hi guys.
Neither "minimize=false" nor "-removeRedundantAttributes" works for me: **type="text"** disappears in each production build, where I use UglifyJsPlugin.

Here is my code:
``` javascript 
 html:   {
        test:    /\.html$/,
        exclude: /node_modules/,
        loader:  'html?minimize=false' // or 'html?-removeRedundantAttributes'
    },
...
uglifyJs: () => {
        return new webpack.optimize.UglifyJsPlugin({
            mangle: false,
            compress: {
                warnings: false
            }
        });
    },
```

Versions:
**css-loader**: 0.23.0
**html-loader**: 0.4.0
**webpack**: 1.12.1

**Comment by jhnns** [comment]
Should be fixed with `html-loader@0.4.2`
